src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/PreferJaggedArraysOverMultidimensionalTests.cs

```diff
@@ -213,6 +214,69 @@ End Class
             GetBasicReturnResultAt(9, 33, "MethodReturningMultidimensionalArray", "Integer(*,*)"));
         }
 
+        [Fact, WorkItem(3650, "https://github.com/dotnet/roslyn-analyzers/issues/3650")]
+        public async Task Method_WhenInterfaceImplementation_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public interface IC
+{
+    int[,] {|#0:MethodReturningMultidimensionalArray|}(int[,] {|#1:array|});
+}
+
+public class C : IC
+{
+    public int[,] MethodReturningMultidimensionalArray(int[,] array)
+        => null;
+}",
+                VerifyCS.Diagnostic(PreferJaggedArraysOverMultidimensionalAnalyzer.ReturnRule).WithLocation(0).WithArguments("MethodReturningMultidimensionalArray", "int[*,*]"),
```

Comment:
@sharwell That's a real pleasure to use the combination markup syntax + diagnostic. It's a lot easier to read and review.

Reply:
Also tagging @carlossanlop who faced a similar issue with an analyzer. It is indeed possible to use markup for analyzer reporting multiple diagnostic descriptors with same ID, and this test is good example of it.